modules/cloud-library/connect/cloud-library.php

```
@@ -126,5 +151,50 @@ public function update_resource( array $template_data ) {
 		return true;
 	}
 
+	public function update_resource_preview( $template_id, $file_data ) {
+		$endpoint = 'resources/' . $template_id . '/preview';
+
+		$boundary = wp_generate_password( 24, false );
+
+		$headers = [
+			'Content-Type' => 'multipart/form-data; boundary=' . $boundary,
+		];
+
+		$body = $this->generate_multipart_payload( $file_data, $boundary, $template_id . '_preview.png' );
+
+		$response = $this->http_request( 'PATCH', $endpoint, [
+			'headers' => $headers,
+			'body' => $body,
+		], [
+			'return_type' => static::HTTP_RETURN_TYPE_ARRAY,
+		] );
+
+		if ( is_wp_error( $response ) || empty( $response['preview_url'] ) ) {
+			return null;
```

Why `null`?

---

We can return an error, but should we display it to the user in some way?